rpcserver.go

```diff
@@ -750,11 +750,11 @@ func (r *rpcServer) addDeps(s *server, macService *macaroons.Service,
 	// TODO(roasbeef): extend sub-sever config to have both (local vs remote) DB
 	err = subServerCgs.PopulateDependencies(
 		r.cfg, s.cc, r.cfg.networkDir, macService, atpl, invoiceRegistry,
-		s.htlcSwitch, r.cfg.ActiveNetParams.Params, s.chanRouter,
-		routerBackend, s.nodeSigner, s.graphDB, s.chanStateDB,
-		s.sweeper, tower, s.towerClient, s.anchorTowerClient,
-		r.cfg.net.ResolveTCPAddr, genInvoiceFeatures,
-		genAmpInvoiceFeatures, getNodeAnnouncement,
+		s.htlcSwitch, r.cfg.ActiveNetParams.Params,
+		s.cc.Wallet.CurrentNumAnchorChans, s.chanRouter, routerBackend,
```

Comment:
We don't need to pass this function in here specifically since we already pass in `s.cc`, we can directly use `cc.Wallet.CurrentNumAnchorChans` in `subserver_config.go`.